controllers/storagecluster/external_ocs.go

```diff
@@ -0,0 +1,284 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	externalClient "github.com/red-hat-storage/ocs-operator/services/provider/client"
+)
+
+var (
+	// externalOCSResources will hold the ExternalResources for storageclusters
+	// ExternalResources can be accessible using the UID of an storagecluster
+	externalOCSResources = map[types.UID][]ExternalResource{}
+)
+
+// isExternalOCSProvider returns true if it is ocs to ocs ExternalStorage consumer cluster
+func isExternalOCSProvider(instance *ocsv1.StorageCluster) bool {
+	return instance.Spec.ExternalStorage.Enable && instance.Spec.ExternalStorage.StorageProviderKind == ocsv1.KindOCS
+}
+
+// getExternalStorageRequestedCapacity returns requestedCapacity,
+// if not set it returns a default value 1Ti
+func getExternalStorageRequestedCapacity(instance *ocsv1.StorageCluster) *resource.Quantity {
+	if instance.Spec.ExternalStorage.RequestedCapacity == nil {
+		q := resource.MustParse("1Ti")
+		return &q
+	}
+	return instance.Spec.ExternalStorage.RequestedCapacity
+}
+
+// newProviderClient returns the *grpc.ClientConn and *providerClient.ConsumerClient
+func newProviderClient(instance *ocsv1.StorageCluster) (*grpc.ClientConn, *externalClient.OCSProviderClient, error) {
+
+	// TODO: implement logic for seprating adress and token
+
+	grpcConnection, err := externalClient.NewGRPCConnection(
+		instance.Spec.ExternalStorage.ConnectionString, []grpc.DialOption{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	consumerClient := externalClient.NewProviderClient(grpcConnection, time.Second*10)
+	return grpcConnection, consumerClient, nil
+}
+
+// onBoardConsumer makes an API call to the provider cluster for onboarding
+func onBoardConsumer(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	clusterVersion := &configv1.ClusterVersion{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "version"}, clusterVersion)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	name := fmt.Sprintf("ocs-consumer-%s", clusterVersion.Spec.ClusterID)
+	response, err := r.externalClient.OnboardConsumer(name, getExternalStorageRequestedCapacity(instance).String())
+	res, err := r.handleOnboardConsumerErrors(instance, err)
+	if err != nil {
+		return res, err
+	}
+
+	if response.StorageConsumerUUID == "" || response.GrantedCapacity == "" {
+		err = fmt.Errorf("External-OCS:OnboardConsumer:response is empty")
+		r.Log.Error(err, "empty response")
+		return reconcile.Result{}, err
+	}
+
+	instance.Status.ExternalStorage.ConsumerID = response.StorageConsumerUUID
+	instance.Status.ExternalStorage.GrantedCapacity = resource.MustParse(response.GrantedCapacity)
+
+	return reconcile.Result{}, nil
+}
+
+// offBoardConsumer makes an API call to the provider cluster for offboarding
+func offBoardConsumer(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
```

Comment:
Lets change to `offboardConsumer` and `onboardConsumer` (`b` in lowercase) everywhere.

Reply:
done